jlm/rvsdg/node.cpp

```diff
@@ -113,7 +113,7 @@ output::remove_user(jlm::rvsdg::input * user)
   {
     if (!node->has_users())
     {
-      JLM_ASSERT(region()->AddBottomNode(*node));
+      region()->AddBottomNode(*node);
```

Comment:
Could we do something like this for all of them instead:

```
auto wasAdded = region()->AddBottomNode(*node);
JLM_ASSERT(wasAdded)
```

Reply:
Doesn't this risk unused variable warnings in release builds instead?

Reply:
I don't think so, as the definition of `JLM_ASSERT` is
```c++
#define JLM_ASSERT(x) \
  do                  \
  {                   \
  } while (0 && (x))
```

Which means the expression in still "used", without being evaluated. Unless the compiler is eliminating dead code before checking, it should not give a warning.

I tested the following in g++ 14.2.1 and clang++ 18.1.8, with `-Wall -Wextra`, and they only complain about `z`.
```c++
int main() {
    int x = 5;
    int y = 7;
    int z = 10;

    do {
    } while(0 & (x));

    return y;
}
```

Reply:
oh right -- changed it accordingly